src/com/google/javascript/jscomp/AbstractCommandLineRunner.java

```diff
@@ -1139,10 +1139,14 @@ void writeOutput(
   @GwtIncompatible("Unnecessary")
   private static void maybeCreateDirsForPath(String pathPrefix) {
     if (!Strings.isNullOrEmpty(pathPrefix)) {
-      String dirName =
-          pathPrefix.charAt(pathPrefix.length() - 1) == File.separatorChar
-              ? pathPrefix.substring(0, pathPrefix.length() - 1)
-              : new File(pathPrefix).getParent();
+      char sep = pathPrefix.charAt(pathPrefix.length() - 1);
+      String dirName;
+      // allow the user to provide any path separator
+      if (sep == '/' || sep == '\\') {
```

Comment:
This was a really good catch.

This method is just badly designed.
IIUC, it's trying to interpret a trailing file separator to mean "create this full path as a directory", but if there isn't a trailing separator it just creates directories up to the last path element.

how confusing.

I've just taken a look at all callers of this function.
All of them pass paths that will be used to create a file except for the one on line 1778

Presumably the idea there is a user may specify either `path/to/my/output/chunks/` or `path/to/my/output/chunks/chunk-` as a prefix

How about lets drop the check for a trailing file separator from this method and instead fix that one call on 1778 to pass a string with a dummy file name appended to eliminate the need for the check?

The parameter to this file then should be named "filePath" instead of "pathPrefix" and the body is much simpler

```
if (!Strings.isNullOrEmpty(filePath)) {
  new File(filePath).getParent().mkdirs();
}
```

Reply:
> How about lets drop the check for a trailing file separator from this method and instead fix that one call on 1778 to pass a string with a dummy file name appended to eliminate the need for the check?

Would this change the behavior of the command line flag `chunk_output_path_prefix`? 
https://github.com/google/closure-compiler/blob/736de5253399c743b439ae03696e9eca3fb86aff/src/com/google/javascript/jscomp/CommandLineRunner.java#L404-L410

https://github.com/theseanl/tscc depends on the behavior of the flag `chunk_output_path_prefix` of treating trailing path separator as a signal for creating a new directory.

Reply:
if i understand @brad4d correctly, it would behave exactly the same (which is the confusing part). if i pass `out/dir/` and pass a dummy file like `out/dir/dummy`, or use `out/dir/mypath` and still use the dummy like `out/dir/mypathdummy`, the result would be exactly the same, the paths would be created up to the last found `/`, so `out/dir`